stdlib/public/Darwin/Accelerate/vImage_CGImageFormat.swift

```diff
@@ -0,0 +1,72 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Accelerate
+
+//===----------------------------------------------------------------------===//
+//
+//  vImage_CGImageFormat
+//
+//===----------------------------------------------------------------------===//
+
+@available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+extension vImage_CGImageFormat {
```

Comment:
If you're going to define `==`, you should conform to `Equatable`. Conforming to `Hashable` as well would be a good idea.

Reply:
The reason it's not `Equatable` is that protocol 'Equatable' requires '==' to be available on OS X 10.9.0 and newer, and we're specifying 9999. The reason it's not `Hashable` is "the components used for hashing must be the same as the components compared in your type’s == operator implementation" - `vImageCGImageFormat_IsEqual` is a black box, so I don't know which properties are used for equivalence. Potentially, I could try something like:

    public func hash(into hasher: inout Hasher) {
        hasher.combine(self.bitmapInfo.rawValue)
        hasher.combine(self.bitsPerComponent)
        hasher.combine(self.bitsPerPixel)
        hasher.combine(self.colorSpace.takeRetainedValue().hashValue)
        hasher.combine(self.componentCount)
        hasher.combine(self.renderingIntent.rawValue)
    }

...but `Hashable` has the same 10.9.0 requirement.

Reply:
@airspeedswift This seems like an appropriate use for @_alwaysEmitIntoClient.

Reply:
If it isn't well-defined what components `==` actually compares, then it makes sense not to implement `Hashable`. (Although: what's the value of providing equality if it's not clear what it does?) The risk here is that the lack of an official `Hashable` conformance is inviting API users to try conforming this type to `Hashable` themselves.

An alternative would be to implement both `==` and `hash(into:)` from scratch here, not calling into `vImageCGImageFormat_IsEqual` at all.

Reply:
@stephentyrone - I have a query outstanding to understand the mechanics of `vImageCGImageFormat_IsEqual`. Let me know if you think it's worth removing this function in the short term and reintroduce it when we have properly understood equivalence and can implement `Hashable`.

Reply:
Yes, let's do that.

Reply:
Super. That's removed and tests pass locally. Cheers!